python/lsst/ts/externalscripts/make_love_stress_tests.py

```diff
@@ -0,0 +1,333 @@
+        self.config = config
+
+        # get credentials
+        self.username = os.environ.get("USER_USERNAME")
+        self.password = os.environ.get("USER_USER_PASS")
+        if self.password is None:
+            raise RuntimeError(
+                "Configuration failed: environment variable USER_USER_PASS not defined"
+            )
+
+        # construct remotes
+        remotes = dict()
+        for name_index in config.data:
+            name, index = salobj.name_to_name_index(name_index)
+            self.log.debug(f"Create remote {name}:{index}")
+            if name_index not in remotes:
+                remote = salobj.Remote(domain=self.domain, name=name, index=index)
+                remotes[name_index] = remote
+        self.remotes = remotes
+
+    async def run(self):
+        """Run script."""
+
+        self.log.info(f"Waiting for {len(self.remotes)} remotes to be ready")
+        await asyncio.gather(*[remote.start_task for remote in self.remotes.values()])
+
+        # Enable all CSCs
+        for remote in self.remotes.values():
+            await salobj.set_summary_state(remote=remote, state=salobj.State.ENABLED)
+
+        event_streams = dict()
+        telemetry_streams = dict()
+        for name_index in self.remotes:
+            name, index = salobj.name_to_name_index(name_index)
+            salinfo = salobj.SalInfo(self.domain, name)
+            try:
+                event_streams[name_index] = salinfo.event_names
+                telemetry_streams[name_index] = salinfo.telemetry_names
+            finally:
+                await salinfo.close()
+
+        # Create clients and listen to ws messages
+        self.log.info(
+            f"Waiting for {self.config.number_of_clients} Manager Clients to be ready"
+        )
+        for i in range(self.config.number_of_clients):
+            client = ManagerClient(
+                self.config.location,
+                self.username,
+                self.password,
+                event_streams,
+                telemetry_streams,
+            )
+            self.clients.append(client)
+            client.create_start_task()
+
+        msg_count = 0
+        while msg_count < self.config.number_of_messages:
+            await asyncio.sleep(self.loop_time_message_collection)
+            for client in self.clients:
+                msg_count += len(client.msg_traces)
+        mean_latency = round(self.get_mean_latency(), 2)
+        self.log.info(
+            f"LOVE stress test result: mean_latency_ms={mean_latency} num_messages={msg_count}"
+        )
+
+    async def cleanup(self):
+        """Return the system to its default status."""
+
+        # Close all ManagerClients
+        for client in self.clients:
+            await client.close()
+
+        # Send all CSCs to Standby
+        for remote in self.remotes.values():
+            await salobj.set_summary_state(remote=remote, state=salobj.State.STANDBY)
+
+        await self.close()
```

Comment:
I suggest removing this call to `self.close()`, unless you have found it to be necessary (in which case I need to know this, as it's a bug). BaseScript should call `close` when needed.

Reply:
I still suggest removing this line.